### PR TITLE
11-Components: SearchTopBar and SearchWidget

### PR DESCRIPTION
**Pull request by DurtDiv3r**
AppTopBar and SearchTopBar components

### CODE REVIEW

File: app/src/main/java/com/islaharper/dawnofandroid/presentation/components/TopBars.kt [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                    contentDescription = stringResource(R.string.search_icon),
+                    tint = MaterialTheme.colorScheme.onPrimary
+                )
+            }
+            IconButton(onClick = {
+                expanded = true
+            }) {
+                Icon(
+                    imageVector = Icons.Default.MoreVert,
+                    contentDescription = "Open Options",
```
Comment (drawers) [inline app/src/main/java/com/islaharper/dawnofandroid/presentation/components/TopBars.kt]: Nit: string resource?

File: app/src/main/java/com/islaharper/dawnofandroid/presentation/components/TopBars.kt [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.islaharper.dawnofandroid.R
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun AppTopBar(
```
Comment (drawers) [inline app/src/main/java/com/islaharper/dawnofandroid/presentation/components/TopBars.kt:32]: Nit: this Composable is missing a modifier

File: app/src/main/java/com/islaharper/dawnofandroid/presentation/components/TopBars.kt [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+}
+
+@Composable
+fun SearchTopBar(
+    text: String,
+    onTextChanged: (String) -> Unit,
+    onSearchClicked: (String) -> Unit,
+    onCloseClicked: () -> Unit
+) {
+    SearchWidget(
```
Comment (drawers) [inline app/src/main/java/com/islaharper/dawnofandroid/presentation/components/TopBars.kt:127]: Where is this component? Is it one you have built yourself? I can't see it in this branch

**Review by drawers — COMMENTED** [review]
Looks okay so far, a few nit picks

Can you please get this branch to a state where it compiles and builds so I can review further? Thanks

File: app/src/main/java/com/islaharper/dawnofandroid/presentation/components/TopBars.kt (hunk 3 above)
Comment (drawers) [reply app/src/main/java/com/islaharper/dawnofandroid/presentation/components/TopBars.kt:127]: Thank you for adding this in, Is it possible to write a preview for this component?

File: app/src/main/java/com/islaharper/dawnofandroid/presentation/components/Widgets.kt [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+                imeAction = ImeAction.Search
+            ),
+            keyboardActions = KeyboardActions(
+                onSearch = {
+                    onSearchClicked(text)
+                    keyboardController?.hide()
+                }
+            ),
+            colors = TextFieldDefaults.textFieldColors(
+                containerColor = Color.Transparent,
```
Comment (drawers) [inline app/src/main/java/com/islaharper/dawnofandroid/presentation/components/Widgets.kt]: Nit: theme color?

File: app/src/main/java/com/islaharper/dawnofandroid/presentation/components/Widgets.kt [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+                cursorColor = MaterialTheme.colorScheme.onPrimary
+            )
+        )
+    }
+}
+
+@Preview(name = "Light")
```
Comment (drawers) [inline app/src/main/java/com/islaharper/dawnofandroid/presentation/components/Widgets.kt]: Thank you for the previews, but like the other comment they don't seem to do light/dark modes.

File: app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -12,5 +12,7 @@
     <string name="sign_in_description">Sign in to continue</string>
     <string name="sign_in_with_google">Sign in with Google</string>
     <string name="please_wait">Please wait...</string>
-
+    <string name="search_icon">Search Icon</string>
```
Comment (drawers) [inline app/src/main/res/values/strings.xml]: Nit: I think this is similar to the comment in the other PR about "Close Icon, Button, double-tap to activate" announcements. Can we please check these?

File: app/src/main/java/com/islaharper/dawnofandroid/presentation/components/Widgets.kt (hunk 5 above)
Comment (DurtDiv3r) [reply app/src/main/java/com/islaharper/dawnofandroid/presentation/components/Widgets.kt]: Morning, I did some experimenting and found a way of displaying the previews correctly. I will implement the change on both PR's.

**Review by drawers — APPROVED** [review]
thank you for persevering with this!

**Pull request merged**